testdata/cheats/Wallet.t.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.18;
+
+import "ds-test/test.sol";
+import "./Vm.sol";
+
+contract Foo {}
+
+contract WalletTest is DSTest {
+    Vm constant vm = Vm(HEVM_ADDRESS);
+
+    function addressOf(uint256 x, uint256 y) internal pure returns (address) {
+        return address(uint160(uint256(keccak256(abi.encode(x, y)))));
+    }
+
+    function testCreateWalletStringPrivAndLabel() public {
+        bytes memory privKey = "this is a priv key";
+        Vm.Wallet memory wallet = vm.createWallet(string(privKey));
+
+        // check wallet.addr against recovered address using private key
+        address expectedAddr = vm.addr(wallet.privateKey);
+        assertEq(expectedAddr, wallet.addr);
+
+        // check wallet.addr against recovered address using x and y coordinates
+        expectedAddr = addressOf(wallet.publicKeyX, wallet.publicKeyY);
+        assertEq(expectedAddr, wallet.addr);
+
+        string memory label = vm.getLabel(wallet.addr);
+        assertEq(label, string(privKey), "labelled address != wallet.addr");
+    }
+
+    function testCreateWalletPrivKeyNoLabel(uint248 pk) public {
```

Comment:
Instead of receiving the private key as type `uint248`, one could receive a private key seed of type `uint256` and create the private key via:
```solidity
uint internal constant Q = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141;

// Let pk ∊ [1, Q).
uint pk = bound(pkSeed, 1, Q-1);
```

This would ensure the whole eligible set of private keys is testable. (Note that `type(uint248).max < Q`, meaning the highest possible private keys are not testable)

Reply:
+1, this would be a stronger test

Reply:
Ha yea that's a good catch! There was an issue with `bound` not being defined, so I fixed that and this nix in this commit here: [8c36615](https://github.com/foundry-rs/foundry/pull/5332/commits/8c36615b811996e8e494e96120d3bc0f79d867ea)